src/gui/TabManager.cc

```diff
@@ -697,15 +697,15 @@ bool TabManager::saveAs(EditorInterface *edt)
 
   if (QFileInfo(filename).suffix().isEmpty()) {
     filename.append(".scad");
```

Comment:
An alternate fix for this issue would be to remove this line, and the manual overwrite confirmation box, entirely. However, that course of action would change the behavior of the "Save As" function. The current behavior is to add `.scad` regardless of whether the user checks the "Automatically select filename extension" box in the Qt window. Even if the user elects not to select an extension and types their filename as `foo2`, this line causes it to be saved as `foo2.scad` anyway. From my investigation, this behavior is why the secondary, manual overwrite confirmation was added in the first place: because without that second box, if a file called `foo2.scad` already existed, it would be overwritten silently because that `.scad` extension is added after Qt has done all its internal checks.